### ISSUE TITLE
Device Independent "Hotkey Special" Bind

### ISSUE THREAD

**Issue by eckso** [libretro/RetroArch#8743]
The goal of the Hotkey Special Bind is to be able to call RetroArch actions with the controller pad, which falls short in button numbers, without the risk of accidentally calling said actions while gaming.

This risk is not associated with keyboards which are not commonly used while playing and offers a much greater array of keys to be directly pressed to perform emulator actions or options.

Currently the Hotkey Special Bind is scope independent (global), meaning if you set it to controller's Guide button, it will also be required to perform Keyboard presses (ie. Guide+p for pausing). While this could make sense from a programming standpoint it defies the original intention of the Hotkey Special and ignores (or repurposes) the original "nul" mapping to something else (Guide button in this case)

In this example Keyboard's "nul" is repurposed as Controller's "Guide":
```
input_enable_hotkey = "nul"
input_enable_hotkey_axis = "nul"
input_enable_hotkey_btn = "Guide"
input_enable_hotkey_mbtn = "nul"
```

**Comment by hizzlekizzle** [comment]
The hotkey is also used on keyboards for classic computer cores that need the full keyboard. That is, so you can use the emulated keyboard without firing off RetroArch hotkey events constantly.

**Comment by eckso** [comment]
Good point, in what cases one needs to trigger a RetroArch event while using the emulated keyboard?
I think some, Screenshot, Savestates, Fullscreen, GUI... but those events live in the non-inputable emulated keyboard keys (F1,F2...) as per my previous [ticket](https://github.com/libretro/RetroArch/issues/8742).
Just thinking on how much of a corner case we are talking.

**Comment by Ferk** [comment]
An example is the PrBoom core.
In Doom ESC opens the menu, and the defaults for quicksave/quickload are F6/F9 as in original Doom. It does not use savestates.

**Comment by soredake** [comment]
Any progress on this?

**Comment by zoltanvb** [comment]
This is nowadays solved by #14728 and #16152
